pytensor/link/pytorch/dispatch/shape.py

```diff
@@ -34,7 +34,7 @@ def shape_i(x):
 def pytorch_funcify_SpecifyShape(op, node, **kwargs):
     def specifyshape(x, *shape):
         assert x.ndim == len(shape)
-        for actual, expected in zip(x.shape, shape):
+        for actual, expected in zip(x.shape, shape, strict=True):
```

Comment:
Even more critical for torch. There's also a risk it would introduce a graph break which we definitely do not want